partner_communication/models/communication_job.py

```diff
@@ -408,21 +419,18 @@ def send(self):
                 job.send_mode = "physical"
                 job.refresh_text()
 
+        # if the call must be done after the sending, an activity is scheduled
+        if self.need_call == "after_sending":
+            self.activity_schedule('mail.mail_activity_data_call',
+                                   summary="call_" + self.need_call,
+                                   user_id=self.user_id.id,
+                                   note="Call {} at (phone) {}"
+                                   .format(self.partner_id.name,
+                                           self.partner_phone))
```

Comment:
You should move this inside the for loop and use `job` instead of `self`. This code will fail if you send multiple communications at once, because you cannot access `self.need_call` on a recordset.

Reply:
This is not yet correct. You must have the loop in the write method.

Reply:
This part of code is inside the send method so I'm not really sure what you mean. Should I use the `job` in `todo.filtered(lambda j: j.send_mode in ("both", "digital"))` or should I use `self` instead ? 

For this PR, I just translated the code to act as if the state call still exist, but create an activity instead. So there might be logic errors before my attempt to do the task and they'll be corrected with this PR.